### PR TITLE
5 create a redux slice for profile

### PR DESCRIPTION
**Pull request by ThaDDDen**

### CODE REVIEW

File: store/store.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { configureStore } from "@reduxjs/toolkit";
 import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
+import { profileReducer } from "./profile/profileSlice";
 
 const store = configureStore({
   reducer: {
+    profile: profileReducer,
     // household: householdReducer
```
Comment (Zaai90) [inline store/store.ts:8]:
```suggestion
```

File: store/profile/profileSlice.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      state.profile = action.payload;
+    });
+    builder.addCase(setProfile.rejected, (state, action) => {
+      state.pending = false;
+      state.error = action.payload || "Unknown error";
+    });
+  },
+});
+
+// export const { deposit, withdrawal } = bankSlice.actions;
```
Comment (Zaai90) [inline store/profile/profileSlice.ts:35]:
```suggestion
```

File: store/profile/profileSlice.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+const setProfile = createAsyncThunk<Profile, Profile, { rejectValue: string }>("profile/setProfile", async (profile, thunkAPI) => {
+  // Api call to set profile
+  try {
+    return profile;
+  } catch (error) {
+    return thunkAPI.rejectWithValue("Error setting profile");
+  }
+});
+
+const test = createAsyncThunk;
```
Comment (Zaai90) [inline store/profile/profileSlice.ts:14]:
```suggestion
```
not in use

File: store/profile/profileSelectors.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import {} from "react-native";
```
Comment (Zaai90) [inline store/profile/profileSelectors.ts:1]:
```suggestion
```
not used

**Review by Zaai90 — CHANGES_REQUESTED** [review]

**Review by 4l3x91 — APPROVED** [review]
Vågar jag approva? Snyggt Tommy! 🐱🐍

File: store/profile/profileState.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export const initialState: ProfileState = {
+  // mock data for now
+  profile: {
+    id: "123",
+    userId: "321",
+    householdId: "456",
+    profileName: "Kattsnake",
+    avatar: "https://steamcdn-a.akamaihd.net/steamcommunity/public/images/avatars/c9/c98a4bd47b02eda8b2acaa1defd5a14a17c07902_full.jpg",
```
[the remark this answers is not in the document]
Comment (ThaDDDen) [reply store/profile/profileState.ts:16]: This is testing data, no need for changes.

File: screens/UserProfileScreen.tsx [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
+  const profile = useAppSelector(selectProfile);
   return (
     <View>
-      <Text>UserProfileScreen</Text>
+      <Text style={{ alignSelf: "center", marginBottom: 30 }}>UserProfileScreen</Text>
+      <Text>Profile Id: {profile.profile.id}</Text>
+      <Text>User Id: {profile.profile.userId}</Text>
+      <Text>Household Id:{profile.profile.householdId}</Text>
+      <Text>Profile name: {profile.profile.profileName}</Text>
+      <Image source={{ uri: profile.profile.avatar }} style={{ width: 80, height: 80 }} />
```
[the remark this answers is not in the document]
Comment (ThaDDDen) [reply screens/UserProfileScreen.tsx:20]: This is testing data, no need for changes.

File: store/profile/profileSlice.ts (hunk 2 above)
Comment (ThaDDDen) [reply store/profile/profileSlice.ts:35]: This stays for now as a reminder to / how to export toolkit reducer actions

**Review by Zaai90 — APPROVED** [review]
w/e

**Pull request merged**